src/test/url-handling.test.js

```diff
@@ -473,6 +474,120 @@ describe('showTabOnly', function() {
   });
 });
 
+describe('ranges', function() {
+  describe('are serializing correctly', () => {
+    it('when there is no range', () => {
+      const { getState } = _getStoreWithURL();
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).not.toContain(`range=`);
+    });
+
+    it('when there is 1 range', () => {
+      const { getState, dispatch } = _getStoreWithURL();
+
+      dispatch(commitRange(1514.587845, 25300));
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).toContain(`range=1514m23786`); // 1.514s + 23786ms
+    });
+
+    it('when rounding down the start', () => {
+      const { getState, dispatch } = _getStoreWithURL();
+
+      dispatch(commitRange(1510.58, 1519.59));
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).toContain(`range=1510m10`); // 1.510s + 10ms
+    });
+
+    it('when the duration is 0', () => {
+      const { getState, dispatch } = _getStoreWithURL();
+
+      dispatch(commitRange(1514, 1514));
+      const urlState = urlStateReducers.getUrlState(getState());
+      const queryString = getQueryStringFromUrlState(urlState);
+      expect(queryString).toMatch(/range=1514000000n(?!0)/); // 1.514s + something non zero
```

Comment:
`(?!0)` means: not followed by a 0, this is pretty standard regexp. I already added a comment to explain it, should I be more explicit about what `(?!0)` does?

Reply:
BTW this is called a negative lookahead assertions. You can find more about this and other assertions in:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Assertions. Look-behind assertions were only recently added to JS regexps, but look-ahead assertions exist for a very long time.

In this case this could be replaced by `[^0]`, which I find a bit less idiomatic :) But if you find it easier to read I'd be happy to do the change.

Reply:
I made this proofread by a friend, he said I should try to unserialize this to see if we can get the value back, instead of checking the result in the URL. I think this is a sensible advice, I'll have a look.

Reply:
For now I left this like this, and added more tests. Waiting for your feedback :-)

Reply:
> (?!0) means: not followed by a 0, this is pretty standard regexp

Yeah, I think even standard regex can be a bit hard at times. I couldn't quite parse the comment on my end either. But now I see what's going on here. In the future I'll try to remember the regex explainer site, but I couldn't google this one when I tried.

Also, is there a problem with just asserting the real value here?

Reply:
The real value isn't guaranteed to stay the same over time, and I didn't want to add more maintenance work to the future developers. Instead, by asserting it's not 0, I feel like we're asserting the right thing.
I'll look into making the comment more explicit.

Reply:
```
      // In the following regexp we want to especially assert that the duration
      // isn't 0. That's why there's this negative look-ahead assertion.
      // Therefore here we're matching a start at 1.514s, and a non-zero
      // duration.
```